source/Halibut/ServiceModel/IMessageRouter.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Halibut.Transport.Protocol;
+
+namespace Halibut.ServiceModel
+{
+    public interface IMessageRouter
```

Comment:
This seems like complexity that adds no real value. e.g. can't this just be

`(proxy as HalibutProxyWithAsync).Configure(SendOutgoingRequest, SendOutgoingRequestAsync, typeof(TService), endpoint, logger, CancellationToken.None);`

Sync is on the removal path eventually so this plumbing to abstract the fact there is a sync and an async version of a method that Async Proxy needs to know about for a medium duration doesn't seem necessary.

Reply:
Removed